pkg/api/v1/router_firmware_set.go

```
@@ -88,6 +94,40 @@ func (r *Router) serverComponentFirmwareSetList(c *gin.Context) {
 	listResponse(c, firmwareSets, pd)
 }
 
+func (r *Router) serverComponentFirmwareSetsSelect(c *gin.Context) {
+	appendToParams := func(params []AttributeListParams, keys, op, value string) []AttributeListParams {
+		return append(params, AttributeListParams{
+			Namespace: FirmwareSetAttributeNS,
+			Keys:      []string{keys},
+			Operator:  OperatorComparitorType(op),
+			Value:     strings.ToLower(value),
+		})
+	}
+
+	// Reuse ComponentFirmwareSetListParams for constructing mods cleaner.
+	var params ComponentFirmwareSetListParams
+	listParams := make([]AttributeListParams, 0)
+	// TODO: support multiple operators(or/and/like) for each query param. May need to
+	// discuss and document the API somewhere.
+	if vendor := c.Query("vendor"); vendor != "" {
```

The vendor, model attributes can be set to be required and are expected as `&vendor=foo&model=bar`. If they are not passed in lets return an error.

The `organization`, `project` are arbitrary labels, so lets accept a URL parameter like `&labels=organization=b,project=c,another=d...`
which is parsed into a `map[string]string` and the selection code will search and return firmware sets that contain those labels exactly,
That way we don't hardcode `organization`, `project` and any new labels.

If there is no additional labels specified, our code should return the firmware set that includes the `default=true` label,
see https://fleet-docs.pages.equinixmetal.net/procedures/firmware-install/#labels-label for reference on how firmware sets are created.

---

Done.
The params is formatted as `vendor=v&model=m&labels=k1=v1,k2=v2,etc`